### ISSUE TITLE
parse outputs changes

### ISSUE THREAD

**Issue by rvoitenko** [containerscrew/tftools#4]
Hi!

not an issue, more a proposal/question.
Do you think it would be difficult to parse "Changes to Outputs:" from the plan as well?
It's also an essential part of terraform plan information.

**Comment by containerscrew** [comment]
Hey @rvoitenko, let me take a look at it and get back to you as soon as I can!

**Comment by containerscrew** [comment]
Hello @rvoitenko Sorry for the time, I haven't been able to review this feature before. Looking at the library, it seems that it would be possible to show the changes in the outputs. I need to research more about it, I will try to get a new release out as soon as possible :)

![image](https://github.com/containerscrew/tftools/assets/131241415/f7f45775-e55b-4aff-82f6-914786735a9e)

**Comment by danielewood** [comment]
@containerscrew You may want to have a look at my current branch:
https://github.com/danielewood/tftools/blob/tags-and-change-summary/

I have this functionality, but its terrible code quality wise. That said, this branch has been in production for the past four months, being used hundreds of times per day on many different repositories.

You may also want to copy a few things from my [bash function](https://github.com/danielewood/tftools/blob/tags-and-change-summary/bash_functions.sh). Terraform plan locks remote state, so if you throw away the plan file, you end up with locked remote state. My bash function addresses this.

**Comment by rvoitenko** [comment]
@containerscrew, @danielewood why is this issue marked as completed?
https://github.com/containerscrew/tftools/pull/6 doesn't consist any logic related to output changes if I understand correctly.

I've tested it:


```
$ terraform plan
Terraform will perform the following actions:

  # local_file.foo will be created
  + resource "local_file" "foo" {
      + content              = "foo!"
      + content_base64sha256 = (known after apply)
      + content_base64sha512 = (known after apply)
      + content_md5          = (known after apply)
      + content_sha1         = (known after apply)
      + content_sha256       = (known after apply)
      + content_sha512       = (known after apply)
      + directory_permission = "0777"
      + file_permission      = "0777"
      + filename             = "./foo.bar"
      + id                   = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + output1 = "test2"
```


```
$ ./tftools version                                                                                                                                                                     ─╯
tftools: v0.9.0 with go version go1.22.2 darwin/arm64
$ ./tftools summarize --show-tags --show-unchanged --compact <plan.json                                                                                                                 ─╯
🟢 Create:
  + local_file.foo
```

**Comment by containerscrew** [comment]
Hi @rvoitenko . You mean imagine you are declaring the following: 

`
output "my_lambda_function_public_url" {
  value = module.mylambda.lambda_function_url
}
`
In this example above, I would be getting the URL of my lambda. (just an example)

https://registry.terraform.io/modules/terraform-aws-modules/lambda/aws/latest?tab=outputs.

So, imagine that in a `terraform plan`, the output of that lambda changes because its URL has changed, do you want it to also be reflected in the `tftools summarize` command?

Have I understood it correctly?

**Comment by rvoitenko** [comment]
> Hi @rvoitenko . Let's see if I understood you correctly.
> 
> Imagine you are declaring the following:
> 
> `output "my_lambda_function_public_url" { value = module.mylambda.lambda_function_url }` In this example above, I would be getting the URL of my lambda. (just an example)
> 
> https://registry.terraform.io/modules/terraform-aws-modules/lambda/aws/latest?tab=outputs.
> 
> Imagine the URL change, so the output of that lambda changes because its URL has changed. Do you want it to also be reflected in the `tftools summarize` command?

Yes, exactly.  Because changes in outputs can be crucial in some cases.

Currently I have to do some dirty hack with bash to add output changes to summary:

```
terraform show --json plan.tfplan > plan.json
cat plan.json | tftools summarize > summary.txt
PLAN_OUTPUT=$(terraform show plan.tfplan -compact-warnings| grep -Ev "You can apply this plan|without changing any real infrastructure")
if echo "$PLAN_OUTPUT" | grep -q "Changes to Outputs:"; then
  echo "🔵 Changes to Outputs:" >> summary.txt
  echo "$PLAN_OUTPUT" | sed -n '/Changes to Outputs:/,$p' | grep -v "Changes to Outputs:" |grep -v '^   '|awk '{print "  +", $2}' >> summary.txt
fi
```

So it would be much nicer if tftools can add output changes to summary.
